blog/2019-08-ng-deploy/README.md

```diff
@@ -0,0 +1,167 @@
+* it is called `deploy` instead of `run` which might be confusing otherwise
+* it throws a meaningfull message in case there is no deployment builder installed (`Cannot find "deploy" target for the specified project. You should add a package that implements deployment capabilities for your favorite platform.`)
+* it has a very short syntax, because by default it will deploy the default project
+* **and the most important point:**
+  It is an invitation to the community to standardise deployments under the umbrella of the Angular CLI 
+
+
+## Command Line Call
+
+Let's take a look at the semantics. How is a deploy builder called?
+
+Without any further arguments, the CLI will try to execute the deploy builder for the default project:
+
+```bash
+ng d [options]
+```
+
+Which is the same as:
+
+```bash
+ng deploy [options]
+```
+
+As you might know, you can manage several Angular Apps (projects) in one workspace.
+The CLI accepts an optional project name, as specified in the `angular.json` configuration file.
+
+```bash
+ng run [PROJECT_NAME]:deploy [options]
+```
+
+The options are optional.
+They will be passed to the deploy target for the selected project. 
+So if you have created a project by calling `ng new your-angular-project`, the complete command could be:
+
+```bash
+ng run your-angular-project:deploy 
+```
+
+
+## Existing providers
+
+There is already a good adaptation for `ng deploy`.
+This is because Minko has [provided a sample project](https://github.com/mgechev/ngx-gh) and helped to implement some deploy targets.
+The following providers are currently available:
+
+* `@angular/fire` (deploys to Firebase)
+* `@azure/ng-deploy` (deploys to Azure Static Hosting)
+* `@netlify-builder/deploy` (deploys to Netlify)
+* `@zeit/ng-deploy` (deploys to Now)
+* `angular-cli-ghpages` (deploys to GitHub pages)
+
+In future you should be able to find more providers via the following search:  
+https://www.npmjs.com/search?q=ng%20deploy
+
+You can try them out by executing the following commands.
+
+1. Install the latest version of the Angular CLI.
+
+   ```sh
+   npm install -g @angular/cli
+   ```
+
+2. Run `ng version`, to make sure you have installed Angular CLI v8.3.0 or greater.
+
+3. Update your existing project using the command:
+
+   ```sh
+   ng update @angular/cli @angular/core
+   ```
+
+4. Add one of the listed providers to your project.
+
+   ```sh
+   ng add [provider]
+   ```
+
+5. Deploy your project via:
+
+   ```sh
+   ng deploy
+   ```
+
+One example:
+
+![angular-cli-ghpages-deploy](angular-cli-ghpages-deploy.gif)
+
+## Caveats
+
+Right now `ng deploy` only support a single deploy target for a project.
+If we want to deploy a project to more than one provider, it is necessary to rewrite the `angular.json` file.
+
+
+<!-- question to Minko! -->
+
+At least for me, is not yet clear whether a deploy builder should compile the project itself or not.

```

Comment:
To reduce the chances to deploy corrupted assets, it's important to build the app right before deploying it.